### PR TITLE
Output error fix 

### PR DESCRIPTION
**Pull request by swilliamson7**
I think this fixes that output error I was getting before. I basically changed the function `get_run_id_path` to return a `run_id` and a `run_path` in the same way that SpeedyWeather does it like you suggested. It seems to be working for me now but you should probably still check, just in case

### CODE REVIEW

File: src/output.jl [hunk 1]
```diff
@@ -172,41 +172,32 @@ end
...
+        runlist = filter(x->endswith(  x,pattern),runlist)
+        existing_runs = [parse(Int,id[5:end]) for id in runlist]
+
+        # get the run id from existing folders
         if length(existing_runs) == 0           # if no runfolder exists yet
-            runpath = joinpath(outpath,"run0000")
-            mkdir(runpath)
-            return 0,runpath
-        else                                    # create next folder
-            if get_id_mode == "fill"  # find the smallest gap in runfolders
```
Comment (milankl) [inline src/output.jl:185]: Note that you removed the `get_id_mode` options. This is fine with me, I only used them once in order to get 200 runs where some failed and I wanted to be able to fill in the gaps or predetermine the output id

File: src/default_parameters.jl [hunk 2]
```diff
@@ -112,7 +112,7 @@
 
     # INITIAL CONDITIONS
     initial_cond::String="rest"         # "rest" or "ncfile" for restart from file
-    initpath::String=outpath            # folder where to pick the restart files from
+    path::String=outpath            # folder where to pick the restart files from
     init_run_id::Int=0                  # run id for restart from run number
     init_starti::Int=-1                 # timestep to start from (-1 meaning last)
     get_id_mode::String="continue"      # How to determine the run id: "continue" or "fill"
```
Comment (milankl) [inline src/default_parameters.jl:118]: If you remove the `get_id_mode` ifs in `output.jl` then we should also remove this field here

File: src/output.jl [hunk 3]
```diff
@@ -172,41 +172,32 @@ end
...
 
-    @unpack output,outpath,get_id_mode = S.parameters
+    @unpack output,outpath,get_id_mode,path = S.parameters
 
     if output
-        runlist = filter(x->startswith(x,"run"),readdir(outpath))
-        existing_runs = [parse(Int,id[4:end]) for id in runlist]
+        
+        pattern = r"run_\d\d\d\d"               # run_???? in regex
+        runlist = filter(x->startswith(x,pattern),readdir(path))
```
Comment (milankl) [inline src/output.jl:180]: Why is this not `outpath` anymore? If you rename this to `path` here then this refers to what was used to be the `initpath` in `default_parameters.jl`?

File: src/default_parameters.jl [hunk 4]
```diff
@@ -112,7 +112,7 @@
 
     # INITIAL CONDITIONS
     initial_cond::String="rest"         # "rest" or "ncfile" for restart from file
-    initpath::String=outpath            # folder where to pick the restart files from
+    path::String=outpath            # folder where to pick the restart files from
```
Comment (milankl) [inline src/default_parameters.jl:115]: Why did you rename this? And use it as the output path?

**Comment by milankl** [comment]
I've also removed the `Manifest.toml`, I know that was in the folder before, but that shouldn't be commited, hence my changes to the `.gitignore` file. Still the CI fails for some reason

**Comment by milankl** [comment]
`ERROR: LoadError: LoadError: libnetcdf not properly installed. Please run Pkg.build("NetCDF")` ouuuuh?

File: src/output.jl (hunk 3 above)
Comment (swilliamson7) [reply src/output.jl:180]: This at least wasn't me renaming something. ShallowWaters.jl had a single function `get_run_id_path` that returned (if I understood correctly) the run_id and the run_path. In SpeedyWeather.jl this was accomplished in two functions`get_run_id` and `create_output_folder` (technically three if you count `run_id_to_string`, which also happens in the single function in ShallowWaters). All I really did was combine the functions from SpeedyWeather into the single function in ShallowWaters by cutting and pasting, so I didn't really think too hard on what the variable names were

File: src/default_parameters.jl (hunk 4 above)
Comment (swilliamson7) [reply src/default_parameters.jl:115]: Because the SpeedyWeather functions just called it path, but I might have misunderstood what `path` in SpeedWeather was versus ShallowWaters initpath?

Comment (swilliamson7) [reply src/default_parameters.jl:115]: I see now that that is what happened, I missed that there was both `outpath` and `path` in Parameters

**Comment by swilliamson7** [comment]
> `ERROR: LoadError: LoadError: libnetcdf not properly installed. Please run Pkg.build("NetCDF")` ouuuuh?

I've had this happen to me before with other functions, I have no idea why. Sometimes it'll tell me something is wrong with SparseArrays for example

File: src/default_parameters.jl (hunk 4 above)
Comment (swilliamson7) [reply src/default_parameters.jl:115]: Sorry about that

File: src/output.jl (hunk 3 above)
Comment (swilliamson7) [reply src/output.jl:180]: See below comments, I now understand what you're referring to

**Comment by swilliamson7** [comment]
This output function still works for me, and fixed the mess up with `path`, 

```julia
function get_run_id_path(S::ModelSetup)

    @unpack output,outpath,get_id_mode = S.parameters

    if output
        
        pattern = r"run_\d\d\d\d"               # run_???? in regex
        runlist = filter(x->startswith(x,pattern),readdir(outpath))
        runlist = filter(x->endswith(  x,pattern),runlist)
        existing_runs = [parse(Int,id[5:end]) for id in runlist]

        # get the run id from existing folders
        if length(existing_runs) == 0           # if no runfolder exists yet
            run_id = 1                          # start with run_0001
        else
            run_id = maximum(existing_runs)+1   # next run gets id +1
        end
        
        id = @sprintf("%04d",run_id)

        run_id2 = string("run_",id)
        run_path = joinpath(outpath,run_id2)
        @assert !(run_id2 in readdir(outpath)) "Run folder $run_path already exists."
        mkdir(run_path)             # actually create the folder

        return run_id, run_path

    else
        return 0,"no runpath"
    end
    
end
```

**Comment by milankl** [comment]
Note sure why there's a precompile error with `libnetcdf`. Can you check whether the tests pass locally?
